third_party/nvfuser/csrc/python_frontend/fusion_definition.h

```diff
@@ -59,17 +61,23 @@ struct TORCH_CUDA_CU_API Tensor {
   //! A unique index to identifiy each recorded state item.
   size_t index;
   size_t dims;
+
+  //! Pointer to the FusionDefinition used to create this tensor
+  FusionDefinition* fd;
```

Comment:
Should I rename these to `fusion_definition`?

Reply:
Note: this is my coding nitpick:

I'm a little bit worried about having `fd` here would make our thread safety refactor later even more complicated/slow, but that could be me being paranoid.

Are we supposed to modify the owner fusion definition through its tensor? is that also why we added the `complete` function as well? I'm wondering if this could be worked around with passing the pointer along with tensors. I have to admit that does feel cumbersome as well.

Reply:
OK, I see where this is coming from now, but I'm holding the same concern :)

I'm ok with us moving forward with this PR on this. cc'ing @kevinstephano for a second opinion.
Meanwhile, we should discuss the feasibility of wrapping tensor methods in python, instead of awkwardly doing it in pybind. IMHO, user experience would be better with easily accessible python scripts, vs going over pybind registration in python_bindings.cpp.

Reply:
I agree it'd be easier if we could implement all this in Python. See my other comment though; I just couldn't find a non-awkward way to do the wrapping and still get the expected types as return values from ops.

Holding the `fd` (will rename to `fusion_definition`) was purely a coincidence thing. If we wrap val types in Python, then we could hold it there instead.